tests/tt_eager/python_api_testing/sweep_tests/tt_lib_ops.py

```diff
@@ -98,6 +98,24 @@ def clone(
     return tt2torch_tensor(t1)
 
 
+@setup_host_and_device
+def typecast(
+    x,
+    *args,
+    device,
+    tt_output_dtype,
+    layout,
+    input_mem_config,
+    output_mem_config,
+    **kwargs,
+):
+    t0 = setup_tt_tensor(x, device, layout[0], input_mem_config[0], tt_output_dtype[0])
+
+    t1 = ttl.tensor.typecast(t0, tt_output_dtype[0], output_mem_config=output_mem_config)
```

Comment:
This still does not test typecast.

This is making the input and output tensor the same dtype so no conversion is taking place. input and output should have their own dtype value